sys/include/net/netif.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netif   Network interfaces
+ * @ingroup     net
+ * @brief       Common network interface API
+ *
+ * This allows access to network interfaces regardless of the network stack
+ * implementation. @anchor NETIF_INVALID The network stack must provide
+ *
+ * - Both a definition for the type `netif_t` and the
+ *   value `NETIF_INVALID` of type `netif_t` in a file `netif_types.h` and
+ * - implementation of all the functions defined in @ref net/netif.h
+ *
+ * @{
+ *
+ * @file
+ * @brief   Common network interface API definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @author  Kaspar Schleiser <kaspar@schleiser.de>
+ */
+#ifndef NETIF_H_
+#define NETIF_H_
+
+#include "net/netopt.h"
+
+#include "netif_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Maximum length for an interface name
+ */
+#ifndef NETIF_NAMELENMAX
+#define NETIF_NAMELENMAX    (8U)
+#endif
+
+/**
+ * @brief   Iterator for the interfaces
+ *
+ * Returns interface after @p last. To start use `last == NETIF_INVALID`.
+ *
+ * @note    Supposed to be implemented by the networking module
+ *
+ * @return next network interface.
+ * @return @ref NETIF_INVALID, if there is no interface after @p last
+ */
+netif_t netif_iter(netif_t last);
+
+/**
+ * @brief   Gets name of an interface
+ *
+ * @pre `name != NULL`
+ * @pre name holds at least @ref NETIF_NAMELENMAX characters
+ *
+ * @note    Supposed to be implemented by the networking module
+ *
+ * @param[in] netif A network interface.
+ * @param[out] name The name of the interface. Must not be `NULL`. Must at least
+ *                  hold @ref NETIF_NAMELENMAX bytes.
+ *
+ * @return  length of @p name on success
+ * @return  0, if @p netif was not a valid interface.
+ */
+int netif_get_name(netif_t netif, char *name);
+
+/**
+ * @brief   Gets interface by name
+ *
+ * @pre `name != NULL`
+ *
+ * @note    Supposed to be implemented by the networking module
+ *
+ * @param[in] name  The name of an interface. Must not be `NULL`.
```

Comment:
Here again I would expect a note about null-termination. The GNRC specific function doesn't need this, e.g., but another network stack may depend on it.